src/Command/ArchiveIdeasCommand.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Console\Attribute\AsCommand;
+use App\Entity\Idea;
+use DateTime;
+
+#[AsCommand(name: 'app:archive-ideas')]
+class ArchiveIdeasCommand extends Command
+{
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        parent::__construct();
+        $this->entityManager = $entityManager;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $currentDate = new DateTime();
+
+        $repository = $this->entityManager->getRepository(Idea::class);
+        $ideas = $repository->findAll();
+
+        foreach ($ideas as $idea) {
+            $endDate = $idea->getEndDate();
+
+            if ($endDate->format('Y-m-d') <= $currentDate->format('Y-m-d')) {
+                $idea->setArchived(true);
+            }
```

Comment:
Do not compare formatted Date, just compare Date.
Cause when you compare two string : 
2023-5-2 > 2023-5-10